circuit/doc.go

```diff
@@ -0,0 +1,152 @@
+requests in the half-open state are accepted concurrently. If any of the requests during the half-open state
+fails, the breaker goes back to open state. If all succeed, it goes to closed state again.
+
+Breaker Type - Failure Rate
+
+The "rate breaker" works similar to the "consecutive breaker", but instead of considering N consecutive failures
+for going open, it maintains a sliding window of the last M events, both successes and failures, and opens only
+when the number of failures reaches N within the window. This way the sliding window is not time based and
+allows the same breaker characteristics for low and high rate traffic.
+
+Usage
+
+When imported as a package, the Registry can be used to hold the circuit breakers and their settings. On a
+higher level, the circuit breaker settings can be simply passed to skipper as part of the skipper.Options
+object, or, equivalently, defined as command line flags.
+
+The following command starts skipper with a global consecutive breaker that opens after 5 failures for any
+backend host:
+
+	skipper -breaker type=consecutive,failures=5
+
+To set only the type of the breaker globally, and the rates individually for the hosts:
+
+	skipper -breaker type=rate,timeout=3m,idleTTL=30m \
+		-breaker host=foo.example.org,window=300,failures=30 \
+		-breaker host=bar.example.org,window=120,failures=45
+
+To enable circuit breakers only for specific hosts:
+
+	skipper -breaker type=disabled \
+		-breaker type=rate,host=foo.example.org,window=300,failures=30
+
+To change (or set) the breaker configurations for an individual route and disable for another, in eskip:
+
+	updates: Method("POST") && Host("foo.example.org")
+	  -> consecutiveBreaker(9)
+	  -> "https://foo.backend.net";
+
+	backendHealthcheck: Path("/healthcheck")
+	  -> disableBreaker()
+	  -> "https://foo.backend.net";
+
+The breaker settings can be defined in the following levels: global, based on the backend host, based on
+individual route settings. The values are merged in the same order, so the global settings serve as defaults for
+the host settings, and the result of the global and host settings serve as defaults for the route settings.
+Setting global values happens the same way as setting host values, but leaving the Host field empty.  Setting
+route based values happens with filters in the route definitions.
+(https://godoc.org/github.com/zalando/skipper/filters/circuit)
+
+Settings - Type
+
+It can be ConsecutiveFailures, FailureRate or Disabled, where the first two values select which breaker to use,
+while the Disabled value can override a global or host configuration disabling the circuit breaker for the
+specific host or route.
+
+Command line name: type. Possible command line values: consecutive, rate, disabled.
+
+Settings - Host
+
+The Host field indicates to which backend host should the current set of settings be applied. Leaving it empty
+indicates global settings.
+
+Command line name: host.
```

Comment:
how would it work? currently if the host is not set, that means global settings (still with individual breaker instances for the hosts)